src/core/api/repository.go

```diff
@@ -363,6 +365,9 @@ func (ra *RepositoryAPI) Delete() {
 		}(t)
 	}
 
+	// publish image delete event
+	ra.buildAndPublishImageDeleteEvent(repoName, tags, project)
```

Comment:
I'm thinking it is still not the way I mentioned in the previous discussions.

Maybe something like:

```go
// Build event first
evt := event.Build(repoName, tags, project)

// Publish the event based on the pkg/notifier.Publish
event.Publish(topic, evt)
```

Reply:
we have considered that.  it is impassable to define an uniform interface for all kinds of events. you can define an interface like `event.Build(repoName, tags, project)`. but there are more events that can not use this interface like an event of modify user password and so on. and event.Build function could only a builder of struct. data should be prepared separately.

Reply:
This is where we need a factory.
And we probably should make event self-explanatory?  i.e. you don't publish a image delete event to "scan complete" topic, so the topic'd better be hidden from the guy who triggers event.
just 
```
event.Publish(evt)
```
not 
```
event.Publish(topic,evt)
```

Reply:
event is common. i don't know how to self-explan a common event to a special one.

Reply:
code fixed, support event.Build & event.Publish